module/vision/BallDetector/src/BallDetector.cpp

```diff
@@ -282,6 +284,24 @@ namespace module::vision {
                     if (log_level <= NUClear::DEBUG || keep) {
                         balls->balls.push_back(std::move(b));
                     }
+
+                    if (horizon.vision_ground_truth.exists) {
+                        Eigen::Affine3f Hcw(horizon.Hcw.cast<float>());
+
+                        const Eigen::Vector3f rBCc = Hcw
+                                                     * Eigen::Vector3f(horizon.vision_ground_truth.rBWw.X,
+                                                                       horizon.vision_ground_truth.rBWw.Y,
+                                                                       horizon.vision_ground_truth.rBWw.Z);
```

Comment:
```suggestion
                        const Eigen::Vector3f rBCc = Hcw * rBWw;
```
should work like this